### PR TITLE
wit apis

### PR DESCRIPTION
**Pull request by nick1udwig**
## Problem

We want to be able to publish APIs as WIT files.

## Solution

TODO

## Docs Update

https://github.com/kinode-dao/kinode-book/pull/198

## Notes

[Sister PR](https://github.com/kinode-dao/kit/pull/143)

### CODE REVIEW

**Comment by dr-frmr** [comment]
Extremely exciting

**Comment by nick1udwig** [comment]
Screenshot from 2024-05-07 14-44-49
Screenshot from 2024-05-07 14-45-02

File: lib/src/core.rs [hunk 1]
```diff
@@ -1082,6 +1082,7 @@ pub struct Erc721Properties {
     pub license: Option<String>,
     pub screenshots: Option<Vec<String>>,
     pub wit_version: Option<(u32, u32, u32)>,
+    pub dependencies: Vec<String>,
```
Comment (dr-frmr) [inline lib/src/core.rs]: Let's make this an `Option<Vec<String>>` and please add the description of this field to the docstring for this type

File: kinode/packages/chess/chess/src/lib.rs [hunk 2]
```diff
@@ -8,27 +8,31 @@ use serde::{Deserialize, Serialize};
...
+//    Resign(String),
+//}
+//
+//#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
+//enum ChessResponse {
+//    NewGameAccepted,
+//    NewGameRejected,
+//    MoveAccepted,
+//    MoveRejected,
+//}
```
Comment (dr-frmr) [inline kinode/packages/chess/chess/src/lib.rs]: can delete this now that it's in wit, maybe add docstring to wit variant

**Review by dr-frmr — COMMENTED** [review]
going to look closer at app store API change, some initial notes

File: lib/src/core.rs (hunk 1 above)
Comment (nick1udwig) [reply lib/src/core.rs]: What is the purpose of having it be an `Option` when an empty vector accomplishes the same thing? JSON parsing optional field?

File: kinode/packages/chess/chess/src/lib.rs (hunk 2 above)
Comment (nick1udwig) [reply kinode/packages/chess/chess/src/lib.rs]: Done in 91717f1

File: lib/src/core.rs (hunk 1 above)
Comment (nick1udwig) [reply lib/src/core.rs]: Changed in 3194b03

Comment (dr-frmr) [reply lib/src/core.rs]: Yeah, if we want this to be backwards-compatible, need to parse successfully when the field doesn't exist in JSON

**Review by dr-frmr — APPROVED** [review]

**Pull request merged**